### DISCUSSION TITLE
How to build in final 'make' step?

### DISCUSSION THREAD

**Discussion by snowman-code** [OpenIDC/mod_auth_openidc#666] [Q&A]
I referred 'INSTALL' document in code files.

I installed all of related tools such as apache, openssl and so on.
After configuration, I ran 'make' command, but result was like this...

`root@ubuntu18:/app/mod_auth/mod_auth_openidc-2.4.9.3# make`
`/usr/lib/apache2/modules  -Wc,"-DNAMEVER=\"mod_auth_openidc-2.4.9.3\" -I/usr/include/x86_64-linux-gnu -I/usr/local/include -I/usr/local/include  -DUSE_MEMCACHE " -Wl,"-lssl -lcrypto -lcurl -L/usr/local/lib -ljansson -L/usr/local/lib -lcjose -ljansson -lcrypto -lpcre  " -Wc,-Wall -Wc,-g -c src/mod_auth_openidc.c src/cache/file.c src/cache/shm.c src/cache/common.c src/oauth.c src/proto.c src/config.c src/util.c src/authz.c src/session.c src/metadata.c src/jose.c src/parse.c src/pcre_subst.c src/cache/memcache.c
/bin/sh: 1: /usr/lib/apache2/modules: Permission denied
Makefile:68: recipe for target 'src/mod_auth_openidc.la' failed
make: *** [src/mod_auth_openidc.la] Error 126`

So, I tried another machine, but I got same error.
I not good at gcc.

I tried
- change mod '/usr/lib/apache2/modules' for 755 and 777.
- change mod '/usr/local/include' for 755 and 777.
- reinstall all tools

Can I get some hint?

Thank you.

**Comment by zandbelt** [comment]
the hint is in the error message "/bin/sh: 1: /usr/lib/apache2/modules: Permission denied", so use "sudo" or similar

**Reply by snowman-code** [reply]
Okay, and I already tried it.
But I got same message.

**Reply by snowman-code** [reply]
I finally found my fault.
I saved source code at private directory and change mod 755 but ii was not working.
I made directory by root and saved source code in it, then 'make' did work good.
Thank you
